src/vcpkg/configuration.cpp

```diff
@@ -1025,17 +1025,21 @@ namespace vcpkg
             return true;
         }*/
 
-        // ([a-z0-9]+(-[a-z0-9]+)*)(\*?)
+        // ^([a-z0-9]+-)*([*]$|[a-z0-9]+($|[*]$))
+        // means that for the non-prefix part: exactly *, or lower-digit with optional *
```

Comment:
The function `bool is_package_pattern(StringView sv)` match that:
* identifier, or
* for the non-prefix part: exactly `*`, or lower-digit with optional `*`

the corresponding regex need combine identifier regex and `^([a-z0-9]+-)*([*]$|[a-z0-9]+($|[*]$))`

So, in json-schema, keep `^([*]|[a-z0-9]+(-[a-z0-9]+)*(-?[*])?)$`, which means that exactly `*`, or identifier with optional `*` or optional `-*`

Reply:
Done

Reply:
Unified as `^([a-z0-9]+-)*([a-z0-9]+[*]?|[*])$`, see https://github.com/microsoft/vcpkg-tool/pull/1429#discussion_r1667217619